src/IW/App/Error.hs

```diff
@@ -126,90 +117,90 @@ data IError
+toHttpError AppError{..} = case appErrorType of
+    NotFound               -> err404
+    ServerError msg        -> err500 { errBody = encodeUtf8 msg }
+    NotAllowed msg         -> err401 { errBody = encodeUtf8 msg }
+    Invalid msg            -> err417 { errBody = encodeUtf8 msg }
+    MissingHeader name     -> err401 { errBody = toLazy $ "Header not found: " <> foldedCase name }
+    HeaderDecodeError name -> err401 { errBody = encodeUtf8 $ "Unable to decode header: " <> name }
+    DbError e              -> err500 { errBody = encodeUtf8 e }
+    DbNamedError e         -> err500 { errBody = show e }
+    GithubHttpError e      -> err500 { errBody = encodeUtf8 e }
+    GithubParseError msg   -> err500 { errBody = encodeUtf8 msg }
+    GithubJSONError msg    -> err500 { errBody = encodeUtf8 msg }
+    GithubUserError msg    -> err500 { errBody = encodeUtf8 msg }
+    UrlDownloadFailed url  -> err500 { errBody = encodeUtf8 $ "Couldn't download file from " <> unUrl url }
 
 ----------------------------------------------------------------------------
 -- Error checks
 ----------------------------------------------------------------------------
 
 isServerError :: AppErrorType -> Bool
-isServerError (InternalError (ServerError _)) = True
-isServerError _                               = False
+isServerError (ServerError _) = True
+isServerError _               = False
 
 isNotAllowed :: AppErrorType -> Bool
-isNotAllowed (InternalError (NotAllowed _)) = True
-isNotAllowed _                              = False
+isNotAllowed (NotAllowed _) = True
+isNotAllowed _              = False
 
 isInvalid :: AppErrorType -> Bool
-isInvalid (InternalError (Invalid _)) = True
-isInvalid _                           = False
+isInvalid (Invalid _) = True
+isInvalid _           = False
 
 ----------------------------------------------------------------------------
 -- Internal Error helpers
 ----------------------------------------------------------------------------
 
 notFound :: AppErrorType
-notFound = InternalError NotFound
+notFound = NotFound
 
 serverError :: Text -> AppErrorType
-serverError = InternalError . ServerError
+serverError = ServerError
 
 notAllowed :: Text -> AppErrorType
-notAllowed = InternalError . NotAllowed
+notAllowed = NotAllowed
 
 invalid :: Text -> AppErrorType
-invalid = InternalError . Invalid
+invalid = Invalid
 
 missingHeader :: HeaderName -> AppErrorType
-missingHeader = InternalError . MissingHeader
+missingHeader = MissingHeader
 
 headerDecodeError :: Text -> AppErrorType
-headerDecodeError = InternalError . HeaderDecodeError
+headerDecodeError = HeaderDecodeError
 
 dbError :: Text -> AppErrorType
-dbError = InternalError . DbError
+dbError = DbError
 
 dbNamedError :: PgNamedError -> AppErrorType
-dbNamedError = InternalError . DbNamedError
+dbNamedError = DbNamedError
 
 githubHttpError :: Text -> AppErrorType
-githubHttpError = GitHubError . HttpError
+githubHttpError = GithubHttpError
```

Comment:
I'm not sure that these helper functions are so helpful now when we inlined the error data type. Do you think we should keep them, or just use the constructors?

Reply:
@vrom911 We should just use constructors imo